### PR TITLE
Bug fix & more tests for bidiag

### PR DESCRIPTION
**Pull request by kshyatt**
The negation method for `Bidiagonal`s would throw an error. I also updated the `show` method to be consistent between upper/lower `Bidiagonal`s.

### CODE REVIEW

File: test/linalg/bidiag.jl [hunk 1]
```diff
@@ -23,6 +23,8 @@ for relty in (Float32, Float64, BigFloat), elty in (relty, Complex{relty})
     debug && println("Test constructors")
     @test Bidiagonal(dv,ev,'U') == Bidiagonal(dv,ev,true)
     @test_throws ArgumentError Bidiagonal(dv,ev,'R')
+    @test_throws DimensionMismatch Bidiagonal(dv,ones(elty,n),true)
+    @test_throws ErrorException Bidiagonal(dv,ev)
```
Comment (tkelman) [inline test/linalg/bidiag.jl]: Many `ErrorExceptions` are red flags that some generic `error` should be replaced by a more specific exception type if possible, I think this case might make the most sense as an `ArgumentError` ?

Comment (kshyatt) [inline test/linalg/bidiag.jl]: Yeah, I think so too. Do you want me to update & resubmit?

Comment (tkelman) [inline test/linalg/bidiag.jl]: may as well? it can be a separate commit since it's not like the current state will fail tests
